macros/graph/PGplot.pl

```diff
@@ -0,0 +1,1914 @@
+		my $tick_align  = ($self->im_y($ymin) - $self->im_y($ypos) < 5)             ? 'bottom' : 'top';
+		my $label_align = ($self->im_y($ypos) - $self->im_y($ymax) < 5)             ? 'top'    : 'bottom';
+		my $label_loc   = $yloc eq 'right' && ($xloc eq 'top' || $xloc eq 'bottom') ? $xmin    : $xmax;
+
+		$self->moveTo($xmin, $ypos);
+		$self->lineTo($xmax, $ypos, 'black', 1.5, 0);
+		$self->draw_label(
+			$xlabel, $label_loc, $ypos,
+			fontsize => 'large',
+			v_align  => $label_align,
+			h_align  => $label_loc == $xmin ? 'left' : 'right'
+		);
+		for my $x (@{ $grid->{xticks} }) {
+			$self->draw_label($x, $x, $ypos, font => 'large', v_align => $tick_align, h_align => 'center')
+				unless ($x == $xpos && $show_y);
+		}
+	}
+	if ($axes->yaxis('visible')) {
+		my $ylabel      = $axes->yaxis('label') =~ s/\\[\(\[\)\]]//gr;
+		my $tick_align  = ($self->im_x($xpos) - $self->im_x($xmin) < 5) ? 'left'                              : 'right';
+		my $label_align = ($self->im_x($xmax) - $self->im_x($xpos) < 5) ? 'right'                             : 'left';
+		my $label_loc   = ($yloc eq 'left' && $xloc eq 'top') || ($yloc eq 'right' && $xloc eq 'top') ? $ymin : $ymax;
+
+		$self->moveTo($xpos, $ymin);
+		$self->lineTo($xpos, $ymax, 'black', 1.5, 0);
+		$self->draw_label(
+			$ylabel, $xpos, $label_loc,
+			fontsize => 'large',
+			v_align  => $label_loc == $ymin ? 'bottom' : 'top',
+			h_align  => $label_align
+		);
+		for my $y (@{ $grid->{yticks} }) {
+			$self->draw_label($y, $xpos, $y, font => 'large', v_align => 'middle', h_align => $tick_align)
+				unless ($y == $ypos && $show_x);
+		}
+	}
+
+	# Draw data a second time to cleanup any issues with the grid and axes.
+	$self->draw_data(2);
+
+	# Print Labels
+	for my $label ($pgplot->data('label')) {
+		$self->draw_label($label->style('label'), $label->x(0), $label->y(0), %{ $label->style });
+	}
+
+	# Draw stamps
+	for my $stamp ($pgplot->data('stamp')) {
+		my $symbol = $stamp->style('symbol');
+		my $color  = $stamp->style('color');
+		my $r      = $stamp->style('radius') || 4;
+		if ($symbol eq 'closed_circle') {
+			$self->draw_circle_stamp($stamp->x(0), $stamp->y(0), $r, $color, 1);
+		} elsif ($symbol eq 'open_circle') {
+			$self->draw_circle_stamp($stamp->x(0), $stamp->y(0), $r, $color);
+		}
+	}
+
+	# Put a black frame around the picture
+	$self->im->rectangle(0, 0, $size->[0] - 1, $size->[1] - 1, $self->color('black'));
+
+	return $pgplot->ext eq 'gif' ? $self->im->gif : $self->im->png;
+}
+
+=head1 AXES OBJECT
+
+This is a hash to store information about the axes (ticks, range, grid, etc)
+with some helper methods. The hash is further split into three smaller hashes:
+
+=over 5
+
+=item xaxis
+
+Hash of data for the horizontal axis.
+
+=item yaxis
+
+Hash of data for the vertical axis.
+
+=item styles
+
+Hash of data for options for the general axis.
+
+=back
+
+=head1 USAGE
+
+The axes object should be accessed through a PGplot object using C<$plot-E<gt>axes>.
```

Comment:
Instead of using the `E<gt>` syntax, make this readable directly with `C<< $plot->axes >>`.  Not the spaces in that are required.   I saw this several other places in this file where this should be fixed as well.